replay_parser/reader.py

```diff
@@ -70,41 +78,37 @@ def read_dict(self) -> Dict:
         """
         result = {}
 
-        while self.peek_byte() != 5:
-            key = self.read_lua()
+        while True:
+            type_ = self.read_byte()
+            if type_ == LUA_END:
+                break
+            key = self.read_lua(type_=type_)
             value = self.read_lua()
             result[key] = value
 
-        self.buffer.read(1)
         return result
 
-    def read_lua(self) -> TYPE_LUA:
+    def read_lua(self, type_: Optional[int] = None) -> TYPE_LUA:
```

Comment:
Why type is optional?

Reply:
Just so that you don't have to always read the type first. Basically so you can still call `self.read_lua()` without the argument and it will know to read 1 byte first to determine the type. Maybe this is confusing and it should be 2 function idk